shared/reducers/chat.js

```diff
@@ -79,6 +79,16 @@ function reducer (state: State = initialState, action: Actions) {
   switch (action.type) {
     case CommonConstants.resetStore:
       return initialState
+    case 'chat:removeOutboxMessage': {
```

Comment:
We didn't seem to have any reducer for removing a message from a thread altogether, so I added one here.